### PR TITLE
Material date picker in data capture library

### LINKED ISSUE #1109: Datacapture, add material date picker to datacapture module.
**Is your feature request related to a problem? Please describe.**

Currently, datacapture module provide date type widget with default platform `DatePickerDialog`, and as of now it does not uses `MaterialDatePicker`, we can refer `DatePickerFragment` class in datacapture.
If it is required to make changes as per figma design, then datacapture must start including material date picker.

Material date picker looks as below.

  
Screen Shot 2022-02-03 at 4 39 31 PM

Screen Shot 2022-02-03 at 4 39 45 PM

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1109 

**Description**
Add material date picker.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**
Screen Shot 2022-03-24 at 12 46 49 PM

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactoryInstrumentedTest.kt [hunk 1]
```diff
@@ -238,6 +241,16 @@ class QuestionnaireItemDatePickerViewHolderFactoryInstrumentedTest {
...
+  @Test
+  fun long_returnDateType() {
+    val localDate = LocalDate.of(2022, Month.JANUARY, 1)
+    val epochMilli = localDate.atStartOfDay().atZone(ZoneId.of("UTC")).toInstant().toEpochMilli()
+
+    assertThat(epochMilli.dateType.day).isEqualTo(1)
+    assertThat(epochMilli.dateType.month).isEqualTo(0)
+    assertThat(epochMilli.dateType.year).isEqualTo(2022)
+  }
+
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactoryInstrumentedTest.kt:253]: why this test case? is this just testing system apis?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt [hunk 2]
```diff
@@ -157,3 +152,9 @@ internal val DateType.localDate
       month + 1,
       day,
     )
+
+internal val Long.localDate: LocalDate
+  get() = Instant.ofEpochMilli(this).atZone(ZoneId.of("UTC")).toLocalDate()
+
+internal val Long.dateType: DateType
+  get() = DateType(localDate.year, localDate.monthValue - 1, localDate.dayOfMonth)
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt:160]: i don't think this makes sense to be an extension function.

having date as a property of a long number is strange, it's not a universal feature. this should be a function.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt [hunk 3]
```diff
@@ -157,3 +152,9 @@ internal val DateType.localDate
       month + 1,
       day,
     )
+
+internal val Long.localDate: LocalDate
+  get() = Instant.ofEpochMilli(this).atZone(ZoneId.of("UTC")).toLocalDate()
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt:157]: i don't think this makes sense to be an extension function. this should be a function.

Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt:157]: actually this function isn't doing the right thing. you shouldn't be able to convert a long to a local date. you're assuming the timezone here and it's not correct. i don't think this function should exist.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt (hunk 2 above)
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt:160]: same here. i don't think this function should exist.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt (hunk 2 above)
Comment (santosh-pingle) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt]: yes, every long can not be converted to DateType.

Done.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt [hunk 4]
```diff
@@ -66,34 +67,34 @@ internal object QuestionnaireItemDatePickerViewHolderFactory :
...
+              .setTitleText(context.getString(R.string.select_date))
+              .setSelection(
+                selectedDate?.atStartOfDay(ZoneId.of("UTC"))?.toInstant()?.toEpochMilli()
+                  ?: MaterialDatePicker.todayInUtcMilliseconds()
+              )
+              .build()
+          datePicker.addOnPositiveButtonClickListener { epochMilli ->
+            textInputEditText.setText(
+              Instant.ofEpochMilli(epochMilli)
+                .atZone(ZoneId.of("UTC"))
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt:82]: extract a constant to be used for date conversion.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Review by kevinmost — APPROVED** [review]

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**